File: frontend/src/theme.ts
export interface ThemeColors {
  foreground: string;
  muted: string;
  accent: string;
  accentAlt: string;
  success: string;
  error: string;
  warning: string;
  info: string;
  border: string;
  dimBorder: string;
  toolCallFg: string;
  planFg: string;
  userFg: string;
  assistantFg: string;
  approvalBorder: string;
  spinner: string;
  agentsPanelBg: string;
  agentsRunning: string;
  agentsDone: string;
}

export interface ThemeConfig {
  name: string;
  colors: ThemeColors;
  spinnerFrames: string[];
  particleChars: string[];
}

const DARK: ThemeConfig = {
  name: 'dark',
  colors: {
    foreground:    '#E2E8F0',
    muted:         '#64748B',
    accent:        '#F97316',
    accentAlt:     '#0EA5E9',
    success:       '#22C55E',
    error:         '#EF4444',
    warning:       '#F59E0B',
    info:          '#38BDF8',
    border:        '#334155',
    dimBorder:     '#1E293B',
    toolCallFg:    '#A78BFA',
    planFg:        '#34D399',
    userFg:        '#93C5FD',
    assistantFg:   '#E2E8F0',
    approvalBorder:'#F59E0B',
    spinner:       '#F97316',
    agentsPanelBg: '#1A1B2E',
    agentsRunning: '#F97316',
    agentsDone:    '#22C55E',
  },
  spinnerFrames: ['⠋','⠙','⠹','⠸','⠼','⠴','⠦','⠧','⠇','⠏'],
  particleChars: ['·','•','◦','∘','○','◌','◎','◉','◈','◆','◇','▪','▫','▸','▹'],
};

const HIGH_CONTRAST: ThemeConfig = {
  name: 'high-contrast',
  colors: {
    foreground:    '#FFFFFF',
    muted:         '#AAAAAA',
    accent:        '#FFFF00',
    accentAlt:     '#00FFFF',
    success:       '#00FF00',
    error:         '#FF4444',
    warning:       '#FFAA00',
    info:          '#00AAFF',
    border:        '#888888',
    dimBorder:     '#444444',
    toolCallFg:    '#FF88FF',
    planFg:        '#88FFAA',
    userFg:        '#88CCFF',
    assistantFg:   '#FFFFFF',
    approvalBorder:'#FFAA00',
    spinner:       '#FFFF00',
    agentsPanelBg: '#1A1A1A',
    agentsRunning: '#FFFF00',
    agentsDone:    '#00FF00',
  },
  spinnerFrames: ['|','/','-','\\'],
  particleChars: ['.','*','+','x','o','0','#','@','%','&'],
};

const CYBER: ThemeConfig = {
  name: 'cyber',
  colors: {
    foreground:    '#E0F0FF',
    muted:         '#4A6A8A',
    accent:        '#00FFD0',
    accentAlt:     '#FF006E',
    success:       '#00FF9F',
    error:         '#FF006E',
    warning:       '#FFD600',
    info:          '#00C8FF',
    border:        '#0D3B5C',
    dimBorder:     '#071E2E',
    toolCallFg:    '#FF006E',
    planFg:        '#00FFD0',
    userFg:        '#00C8FF',
    assistantFg:   '#E0F0FF',
    approvalBorder:'#FFD600',
    spinner:       '#00FFD0',
    agentsPanelBg: '#0A0A0A',
    agentsRunning: '#00FFD0',
    agentsDone:    '#00FF9F',
  },
  spinnerFrames: ['◐','◓','◑','◒'],
  particleChars: ['▓','▒','░','█','▄','▀','■','□','▪','▫'],
};

export const THEMES: Record<string, ThemeConfig> = {
  dark: DARK,
  'high-contrast': HIGH_CONTRAST,
  cyber: CYBER,
};

File: crates/sentinel-ai-tui/src/theme.rs
use ratatui::style::Color;

pub struct ThemeColors {
    pub foreground: Color,
    pub muted: Color,
    pub accent: Color,
    pub accent_alt: Color,
    pub success: Color,
    pub error: Color,
    pub warning: Color,
    pub info: Color,
    pub border: Color,
    pub dim_border: Color,
    pub tool_call_fg: Color,
    pub plan_fg: Color,
    pub user_fg: Color,
    pub assistant_fg: Color,
    pub approval_border: Color,
    pub spinner: Color,
}

pub struct ThemeConfig {
    pub name: &'static str,
    pub colors: ThemeColors,
    pub spinner_frames: &'static [&'static str],
    pub particle_chars: &'static [&'static str],
}

pub fn dark_theme() -> ThemeConfig {
    ThemeConfig {
        name: "dark",
        colors: ThemeColors {
            foreground: Color::Rgb(226, 232, 240),
            muted: Color::Rgb(100, 116, 139),
            accent: Color::Rgb(249, 115, 22),
            accent_alt: Color::Rgb(14, 165, 233),
            success: Color::Rgb(34, 197, 94),
            error: Color::Rgb(239, 68, 68),
            warning: Color::Rgb(245, 158, 11),
            info: Color::Rgb(56, 189, 248),
            border: Color::Rgb(51, 65, 85),
            dim_border: Color::Rgb(30, 41, 59),
            tool_call_fg: Color::Rgb(167, 139, 250),
            plan_fg: Color::Rgb(52, 211, 153),
            user_fg: Color::Rgb(147, 197, 253),
            assistant_fg: Color::Rgb(226, 232, 240),
            approval_border: Color::Rgb(245, 158, 11),
            spinner: Color::Rgb(249, 115, 22),
        },
        spinner_frames: &["⠋", "⠙", "⠹", "⠸", "⠼", "⠴", "⠦", "⠧", "⠇", "⠏"],
        particle_chars: &["·", "•", "◦", "∘", "○", "◌", "◎", "◉", "◈", "◆", "◇", "▪", "▫", "▸", "▹"],
    }
}

pub fn high_contrast_theme() -> ThemeConfig {
    ThemeConfig {
        name: "high-contrast",
        colors: ThemeColors {
            foreground: Color::Rgb(255, 255, 255),
            muted: Color::Rgb(170, 170, 170),
            accent: Color::Rgb(255, 255, 0),
            accent_alt: Color::Rgb(0, 255, 255),
            success: Color::Rgb(0, 255, 0),
            error: Color::Rgb(255, 68, 68),
            warning: Color::Rgb(255, 170, 0),
            info: Color::Rgb(0, 170, 255),
            border: Color::Rgb(136, 136, 136),
            dim_border: Color::Rgb(68, 68, 68),
            tool_call_fg: Color::Rgb(255, 136, 255),
            plan_fg: Color::Rgb(136, 255, 170),
            user_fg: Color::Rgb(136, 204, 255),
            assistant_fg: Color::Rgb(255, 255, 255),
            approval_border: Color::Rgb(255, 170, 0),
            spinner: Color::Rgb(255, 255, 0),
        },
        spinner_frames: &["|", "/", "-", "\\"],
        particle_chars: &[".", "*", "+", "x", "o", "0", "#", "@", "%", "&"],
    }
}

pub fn cyber_theme() -> ThemeConfig {
    ThemeConfig {
        name: "cyber",
        colors: ThemeColors {
            foreground: Color::Rgb(224, 240, 255),
            muted: Color::Rgb(74, 106, 138),
            accent: Color::Rgb(0, 255, 208),
            accent_alt: Color::Rgb(255, 0, 110),
            success: Color::Rgb(0, 255, 159),
            error: Color::Rgb(255, 0, 110),
            warning: Color::Rgb(255, 214, 0),
            info: Color::Rgb(0, 200, 255),
            border: Color::Rgb(13, 59, 92),
            dim_border: Color::Rgb(7, 30, 46),
            tool_call_fg: Color::Rgb(255, 0, 110),
            plan_fg: Color::Rgb(0, 255, 208),
            user_fg: Color::Rgb(0, 200, 255),
            assistant_fg: Color::Rgb(224, 240, 255),
            approval_border: Color::Rgb(255, 214, 0),
            spinner: Color::Rgb(0, 255, 208),
        },
        spinner_frames: &["◐", "◓", "◑", "◒"],
        particle_chars: &["▓", "▒", "░", "█", "▄", "▀", "■", "□", "▪", "▫"],
    }
}

pub fn get_theme(name: &str) -> ThemeConfig {
    match name {
        "dark" => dark_theme(),
        "high-contrast" => high_contrast_theme(),
        "cyber" => cyber_theme(),
        _ => dark_theme(),
    }
}
